src/main/java/Duke.java

```diff
@@ -1,10 +1,167 @@
+import java.lang.System;
+import java.util.Scanner;
+
 public class Duke {
+    private static final String SEPARATOR = "____________________________________________________________";
+    protected static final int MAX_NUM_TASKS_STORED = 100;  // maximum number of stored tasks = 100
+    private static Task[] tasks = new Task[MAX_NUM_TASKS_STORED];
+    private static int inputStoredIndex = 0;
+    enum TaskType {
+        TODO, DEADLINE, EVENT, TASK;
+    }
+
     public static void main(String[] args) {
+        printLogo();
+        greet();
+        echo();
+    }
+
+    public static void printLogo(){
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
     }
+
+    public static void greet(){
+        String greeting = "Hello! I'm Duke\nWhat can I do for you?";
+
+        System.out.println(SEPARATOR);
+        System.out.println(greeting);
+        System.out.println(SEPARATOR);
+    }
+
+    public static void echo(){
+        Scanner scanner = new Scanner(System.in);
+        String userInputSentence;
+
+        while (true) {
+            userInputSentence = scanner.nextLine();
+            executeCommand(userInputSentence);
+        }
+    }
+
+    private static void executeCommand(String userInputSentence) {
+        boolean isEqualToTERMIATION_SIGNAL = userInputSentence.toLowerCase().equals(Commands.TERMINATION_SIGNAL);
+        boolean isEqualToLIST_PRINT_COMMAND = userInputSentence.toLowerCase().equals(Commands.LIST_PRINT_COMMAND);
+        boolean isEqualToSET_TASK_DONE_COMMAND = userInputSentence.length() >= Commands.SET_TASK_DONE_COMMAND.length() &&
+                    userInputSentence.substring(0, Commands.SET_TASK_DONE_COMMAND.length()).equals(Commands.SET_TASK_DONE_COMMAND);
```

Comment:
The statement seems to be a little bit too long and complicated. You can consider shortening it by defining new variables to hold some values

Reply:
Sure, I just found `startWith()` method is a good alternative